Firestore/core/test/firebase/firestore/remote/serializer_test.cc

```diff
@@ -1004,6 +1102,278 @@ TEST_F(SerializerTest, DecodeMaybeDocWithoutFoundOrMissingSetShouldFail) {
+  filter.set_op(v1::StructuredQuery::FieldFilter::LESS_THAN);
+  filter.mutable_value()->set_integer_value(42);
+
+  ExpectRoundTrip(model, proto);
+}
+
+TEST_F(SerializerTest, EncodesMultipleFiltersOnDeeperCollections) {
+  core::Query q =
+      Query("rooms/1/messages/10/attachments")
+          .AddingFilter(Filter("prop", ">=", 42))
+          .AddingFilter(Filter("author", "==", "dimond"))
+          .AddingFilter(Filter("tags", "array_contains", "pending"));
+  QueryData model = CreateQueryData(std::move(q));
+
+  v1::Target proto;
+  proto.mutable_query()->set_parent(
+      "projects/p/databases/d/documents/rooms/1/messages/10");
+  proto.set_target_id(1);
+
+  v1::StructuredQuery::CollectionSelector from;
+  from.set_collection_id("attachments");
+  *proto.mutable_query()->mutable_structured_query()->add_from() =
+      std::move(from);
+
+  v1::StructuredQuery::Filter filter1;
+  v1::StructuredQuery::FieldFilter& field1 = *filter1.mutable_field_filter();
+  field1.mutable_field()->set_field_path("prop");
+  field1.set_op(v1::StructuredQuery::FieldFilter::GREATER_THAN_OR_EQUAL);
+  field1.mutable_value()->set_integer_value(42);
+
+  v1::StructuredQuery::Filter filter2;
+  v1::StructuredQuery::FieldFilter& field2 = *filter2.mutable_field_filter();
+  field2.mutable_field()->set_field_path("author");
+  field2.set_op(v1::StructuredQuery::FieldFilter::EQUAL);
+  field2.mutable_value()->set_string_value("dimond");
+
+  v1::StructuredQuery::Filter filter3;
+  v1::StructuredQuery::FieldFilter& field3 = *filter3.mutable_field_filter();
+  field3.mutable_field()->set_field_path("tags");
+  field3.set_op(v1::StructuredQuery::FieldFilter::ARRAY_CONTAINS);
+  field3.mutable_value()->set_string_value("pending");
+
+  v1::StructuredQuery::CompositeFilter& composite =
+      *proto.mutable_query()
+           ->mutable_structured_query()
+           ->mutable_where()
+           ->mutable_composite_filter();
+  composite.set_op(v1::StructuredQuery::CompositeFilter::AND);
+  *composite.add_filters() = std::move(filter1);
+  *composite.add_filters() = std::move(filter2);
+  *composite.add_filters() = std::move(filter3);
+
+  v1::StructuredQuery::Order order1;
+  order1.mutable_field()->set_field_path("prop");
+  order1.set_direction(v1::StructuredQuery::ASCENDING);
+  *proto.mutable_query()->mutable_structured_query()->add_order_by() =
+      std::move(order1);
+
+  v1::StructuredQuery::Order order2;
+  order2.mutable_field()->set_field_path(FieldPath::kDocumentKeyPath);
+  order2.set_direction(v1::StructuredQuery::ASCENDING);
+  *proto.mutable_query()->mutable_structured_query()->add_order_by() =
+      std::move(order2);
+
+  ExpectRoundTrip(model, proto);
+}
+
+TEST_F(SerializerTest, EncodesNullFilter) {
+  SCOPED_TRACE("EncodesNullFilter");
+  ExpectUnaryOperator(Value(nullptr),
+                      v1::StructuredQuery_UnaryFilter_Operator_IS_NULL);
+}
+
+TEST_F(SerializerTest, EncodesNanFilter) {
+  SCOPED_TRACE("EncodesNanFilter");
```

Comment:
It seems like this is something we could just do in the fixture, using the method in https://github.com/firebase/firebase-ios-sdk/pull/3899/files#diff-9cd97ed5d34722d070c417dea0c7b689R132

Reply:
Bump. This would save needing to write SCOPED_TRACE in every method in here.

Reply:
Looking at it now, it seems like it's incorrect -- `ScopedTrace` constructor takes the file name and line as its first two arguments, which the code in `AsyncTest` hardcodes to the same values? Am I missing something here?